src/www/model/events.ts

```diff
@@ -50,7 +50,7 @@ export class ServerReconnecting implements OutlineEvent {
 export class EventQueue {
   private queuedEvents: OutlineEvent[] = [];
   // tslint:disable-next-line: no-any
-  private listenersByEventType = new Map<string, any[]>();
+  private listenersByEventType = new Map<Function, any[]>();
```

Comment:
Maybe we could define a more specific type than `Function`:

```ts
export type OutlineEventConstructor<T extends OutlineEvent> = {new(...args: any[]): T};
```

We could use it in the `subscribe` method as well.

Reply:
What do we put in for `T` in the map key type?  Also we have to be careful defining this because of [contravariance of function parameters](https://stackoverflow.com/a/58459554/1702373).  It would be wrong to use something like `{new(...args: any[]): OutlineEvent}` for this reason.

Reply:
Good point. I guess it's not feasible, even with an added layer of indirection. It still feels odd to me to key by `Function` and would rather use (arbitrary) strings.